app/src/main/java/org/wikipedia/util/DateUtil.kt

```diff
@@ -156,18 +154,10 @@ object DateUtil {
     }
 
     @JvmStatic
-    fun getYearDifferenceString(year: Int): String {
+    fun getYearDifferenceString(year: Int, languageCode: String): String {
         val diffInYears = Calendar.getInstance()[Calendar.YEAR] - year
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            when (diffInYears) {
-                0 -> RelativeDateTimeFormatter.getInstance().format(RelativeDateTimeFormatter.Direction.THIS, RelativeDateTimeFormatter.AbsoluteUnit.YEAR)
-                1 -> RelativeDateTimeFormatter.getInstance().format(RelativeDateTimeFormatter.Direction.LAST, RelativeDateTimeFormatter.AbsoluteUnit.YEAR)
-                -1 -> RelativeDateTimeFormatter.getInstance().format(RelativeDateTimeFormatter.Direction.NEXT, RelativeDateTimeFormatter.AbsoluteUnit.YEAR)
-                else -> RelativeDateTimeFormatter.getInstance().format(diffInYears.toDouble(), RelativeDateTimeFormatter.Direction.LAST, RelativeDateTimeFormatter.RelativeUnit.YEARS)
-            }
-        } else {
-            val context = WikipediaApp.getInstance().applicationContext
-            if (diffInYears == 0) context.getString(R.string.this_year) else context.resources.getQuantityString(R.plurals.diff_years, diffInYears, diffInYears)
-        }
+        val targetResource = L10nUtil.getResourcesForWikiLang(languageCode) ?: WikipediaApp.getInstance().resources
```

Comment:
This removes the usage of the more powerful `RelativeDateTimeFormatter`, which is the preferred formatter as of Android N.  The formatter has `getInstance()` functions that take a `Locale` parameter.